Dockerfile

```diff
@@ -15,7 +17,11 @@ COPY api/ api/
 COPY internal/controllers/ internal/controllers/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager internal/main.go
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make oci-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} GO111MODULE=on go build -a -o manager internal/main.go
 
 FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
```

Comment:
Without specifying any platform, it should automatically assume ones from `--platform` passed to docker/podman build right?

Reply:
That's what seemed to happen when I was working on https://github.com/cryostatio/cryostat/pull/1352 . The correct base image got automatically pulled and built on top of, both when I did it manually with podman as well as when I did it through the jib plugin.

Reply:
Right thanks! No need any updates here then.

Reply:
I was able to confirm this by adding `-o type=tar,dest=./test.tar` to the `docker buildx` command, then extracting the manager binary from the tarball:
```
$ file manager 
manager: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=s9qHdPIc1o_3I_4lPFO_/Y6beLdq3XQXAY2UZSJzf/gltfY-CbCYZBYIrTAmmM/E5RDVvqMGdi8cBTOUIyX, with debug_info, not stripped
```

Reply:
Niceee! Thanks again!